src/connectors/infinite-hits/connectInfiniteHits.js

```diff
@@ -130,6 +151,22 @@ export default function connectInfiniteHits(renderFn, unmountFn) {
       dispose() {
         unmountFn();
       },
+
+      getWidgetState(uiState, { searchParameters }) {
```

Comment:
We've implemented the `routing` but we always run it even when `showPrevious` is disabled. It means that previous implementation that use the widget have now their URL sync with it. We should only enable it when the previous button is present otherwise we might "break" their implementation.